src/app/clusters/icd-management-server/icd-management-server.cpp

```diff
@@ -126,7 +127,7 @@ CHIP_ERROR IcdManagementAttributeAccess::ReadRegisteredClients(EndpointId endpoi
 
                 Structs::MonitoringRegistrationStruct::Type s{ .checkInNodeID    = e.checkInNodeID,
                                                                .monitoredSubject = e.monitoredSubject,
-                                                               .key              = e.key,
+                                                               .key              = chip::ByteSpan(), // Do not expose the key
```

Comment:
So this violates the spec as currently written.  Can we please make sure that the spec is at least somewhat moving in this direction (e.g. issue has been discussed in a TT, there is some sort of consensus, maybe a spec PR exists) before making SDK changes that mis-align with the spec?

Reply:
That's actually my next task... 😄

Reply:
Change is needed to implement a second part of the specs, so it's really just a matter of jungling priorities at this moment.

Reply:
@bzbarsky-apple Keep in mind that the impact of this changes only applies to LIT ICD which are unsupported for now both in the spec for 1.2 and in the SDK.

Reply:
https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7543